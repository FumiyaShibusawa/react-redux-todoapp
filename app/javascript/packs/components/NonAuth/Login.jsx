import * as React from "react"

class Login extends React.Component {
  render() {
    return(
      <div className="login-form">
        <h1>Login</h1>
        <form action="/sessions" method="POST" charSet="UTF-8">
          <input name="authenticity_token" type="hidden"/>
          <input name="user[name]" type="text" placeholder="Enter your name" autoComplete="username"/>
          <input name="session[email]" type="text" placeholder="Enter your email" autoComplete="username"/>
          <input name="session[password]" type="password" placeholder="Enter your password" autoComplete="current-password"/>
          <button type="submit">Login</button>
        </form>
      </div>
    )
  }
}


export default Login
