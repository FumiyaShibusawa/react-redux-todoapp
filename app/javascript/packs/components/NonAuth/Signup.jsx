import * as React from "react"

class SignUp extends React.Component {
  render() {
    return(
        <div className="signup-form">
          <h1>Register</h1>
          <form onSubmit={ (e) => {
            e.preventDefault();
            this.props.handleSignUp(e);
          } }>
            <input name="email" type="text" placeholder="Enter your email" autoComplete="username"/>
            <input name="password" type="password" placeholder="Enter your password" autoComplete="current-password"/>
            <button type="submit">Register</button>
          </form>
          <div className="goto-login" data-modal="popup">Already have your account?</div>
        </div>
    )
  }
}

export default SignUp
