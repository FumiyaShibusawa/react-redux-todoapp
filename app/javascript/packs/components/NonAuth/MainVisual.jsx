import * as React from "react"
import Signup from "./Signup"
import Login from "./Login"

const MainVisual = () => (
  <div>
    <div className="modal">
      <Login/>
    </div>
    <div className="main-visual">
      <div className="inner">
        <div className="message">
          <h2>Simple Todo App</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum</p>
        </div>
        <Signup/>
      </div>
    </div>
    <div className="features">
      <div className="inner">
        <h2>This sample is built with...</h2>
        <div className="features-grid">
          <section>
            <h3>React</h3>
            <img src="assets/react.svg"/>
            <p>UI view built solely with React components.</p>
          </section>
          <section>
            <h3>Redux</h3>
            <img src="assets/redux.svg"/>
            <p>UI view built solely with React components.</p>
          </section>
          <section>
            <h3>JWT</h3>
            <img src="assets/jwt.svg"/>
            <p>UI view built solely with React components.</p>
          </section>
        </div>
      </div>
    </div>
    <div className="contribute">
      <section>
        <h2>Give a PR to improve this sample!</h2>
        <a href="https://github.com/FumiyaShibusawa/react-redux-todoapp" target="_blank">
          https://github.com/FumiyaShibusawa/react-redux-todoapp
        </a>
      </section>
    </div>
  </div>
)

export default MainVisual
