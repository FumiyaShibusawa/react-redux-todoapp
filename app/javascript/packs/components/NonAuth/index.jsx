import * as React from "react"
import MainVisual from "./MainVisual"

class NonAuth extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div className="wall-paper">
        <main className="wrapper">
          <MainVisual/>
        </main>
      </div>
    )
  }
}

export default NonAuth
