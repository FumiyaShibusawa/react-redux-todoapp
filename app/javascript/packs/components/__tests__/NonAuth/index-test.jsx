import * as React from 'react';
import NonAuth from '../../NonAuth';
import renderer from 'react-test-renderer';

test('it renders correctly', () => {
  const tree = renderer.create(
    <NonAuth />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
