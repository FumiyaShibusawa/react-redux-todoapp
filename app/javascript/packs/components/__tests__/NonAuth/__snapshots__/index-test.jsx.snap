// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="wall-paper"
>
  <main
    className="wrapper"
  >
    <div
      className="modal"
    >
      <div
        className="modal-overlay"
      />
      <div
        className="login-form"
      >
        <h1>
          Login
        </h1>
        <form
          action="/sessions"
          charSet="UTF-8"
          method="POST"
        >
          <input
            name="authenticity_token"
            type="hidden"
          />
          <input
            autoComplete="username"
            name="session[name]"
            placeholder="Enter your name"
            type="text"
          />
          <input
            autoComplete="email"
            name="session[email]"
            placeholder="Enter your email"
            type="text"
          />
          <input
            autoComplete="current-password"
            name="session[password]"
            placeholder="Enter your password"
            type="password"
          />
          <button
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
    <div
      className="main-visual"
    >
      <div
        className="inner"
      >
        <div
          className="message"
        >
          <h2>
            Simple Todo App
          </h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
          </p>
        </div>
        <div
          className="signup-form"
        >
          <h1>
            Register
          </h1>
          <form
            action="/users"
            charSet="UTF-8"
            method="POST"
          >
            <input
              name="authenticity_token"
              type="hidden"
            />
            <input
              autoComplete="username"
              name="user[name]"
              onChange={[Function]}
              placeholder="Enter your name"
              type="text"
            />
            <input
              autoComplete="email"
              name="user[email]"
              placeholder="Enter your email"
              type="text"
            />
            <input
              autoComplete="current-password"
              name="user[password]"
              placeholder="Enter your password"
              type="password"
            />
            <button
              type="submit"
            >
              Register
            </button>
          </form>
          <div
            className="goto-login"
            data-modal="popup"
          >
            Already have your account?
          </div>
        </div>
      </div>
    </div>
    <div
      className="features"
    >
      <div
        className="inner"
      >
        <h2>
          This sample is built with...
        </h2>
        <div
          className="features-grid"
        >
          <section>
            <h3>
              React
            </h3>
            <img
              src="assets/react.svg"
            />
            <p>
              UI view built solely with React components.
            </p>
          </section>
          <section>
            <h3>
              Redux
            </h3>
            <img
              src="assets/redux.svg"
            />
            <p>
              State management for todolists with Redux.
            </p>
          </section>
          <section>
            <h3>
              JWT
            </h3>
            <img
              src="assets/jwt.svg"
            />
            <p>
              Token-based Authentication on asynchronous requests.
            </p>
          </section>
        </div>
      </div>
    </div>
    <div
      className="contribute"
    >
      <section>
        <h2>
          Give a PR to improve this sample!
        </h2>
        <a
          href="https://github.com/FumiyaShibusawa/react-redux-todoapp"
          target="_blank"
        >
          https://github.com/FumiyaShibusawa/react-redux-todoapp
        </a>
      </section>
    </div>
  </main>
</div>
`;
